base/refpointer.jl

```diff
@@ -136,12 +141,41 @@ unsafe_convert(::Type{Ptr{T}}, r::Ref{NTuple{N,T}}) where {N,T} =
 unsafe_convert(::Type{Ptr{T}}, r::Ptr{NTuple{N,T}}) where {N,T} =
     convert(Ptr{T}, r)
 
-###
+### Methods for a Ref object that is backed by a object field
+"""
+    RefField <: Ref{T}
+
+Create an object reflection adaptor that represents the properties of a specific object field.
+"""
+struct RefField{f,T,S} <: Ref{T}
+    x::RefValue{S}
+    RefField{f,T,S}(x::S) where {f,T,S} = new{f::Int,T,S}(RefValue(x))
+end
```

Comment:
How does this work with immutables?  For example, if I have:

```julia
x = (a=1, b=2)
ra = RefField(x, :a)
rb = RefField(x, :b)
ra[] = 10
rb[] = 20
```

After this code, I suppose `x === (a=1, b=2)` still holds?  How do I get `(a=10, b=20)`?  Is there going to be some compiler intervention?

Reply:
Is this similar to C's address operator `&x.a`?